### ISSUE TITLE
torch model.restore() error on CUDA mismatch

### ISSUE THREAD

**Issue by paulsonak** [deepchem/deepchem#2956]
## 🐛 Bug

If a model is trained with a GPU but it's trying to reload the model when GPU isn't available (for example on a different machine), torch.load() fails unless the device is explicitly specified as 'cpu'.

## To Reproduce

1. Train torch model on GPU
2. Transfer model to non-GPU computer
3. Try to load the model with torch_model.restore()

[torch_restore_stack_trace.txt](https://github.com/deepchem/deepchem/files/8943790/torch_restore_stack_trace.txt)

## Expected behavior
Even if GPU isn't available, the model should be able to load on a different machine. 

## Environment

* OS: Red Hat Enterprise Linux Server release 7.9 (Maipo)
* Python version: 3.7.10
* DeepChem version: 2.6.1
* RDKit version (optional): 2022.03.01
* TensorFlow version (optional): 2.8.0
* PyTorch version (optional): 1.7.0
* Any other relevant information: CUDA 11.1.0 was used to train the model

## Additional context

I edited `/deepchem/models/torch_models/torch_model.py restore() `to the following and was able to resolve the problem, but the `cuda:0` part should probably be changed to something modular.
```
  def restore(self,
              checkpoint: Optional[str] = None,
              model_dir: Optional[str] = None) -> None:
    """Reload the values of all variables from a checkpoint file.

    Parameters
    ----------
    checkpoint: str
      the path to the checkpoint file to load.  If this is None, the most recent
      checkpoint will be chosen automatically.  Call get_checkpoints() to get a
      list of all available checkpoints.
    model_dir: str, default None
      Directory to restore checkpoint from. If None, use self.model_dir.  If
      checkpoint is not None, this is ignored.
    """
    self._ensure_built()

    ###
    device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
    ###

    if checkpoint is None:
      checkpoints = sorted(self.get_checkpoints(model_dir))
      if len(checkpoints) == 0:
        raise ValueError('No checkpoint found')
      checkpoint = checkpoints[0]

    ###
    data = torch.load(checkpoint, map_location=device)
    ###

    self.model.load_state_dict(data['model_state_dict'])
    self._pytorch_optimizer.load_state_dict(data['optimizer_state_dict'])
    self._global_step = data['global_step']
```

**Comment by rbharath** [comment]
Thank you for the bug report! I'll mark this as a bug and open for contribution

**Comment by advikavs** [comment]
I tried reproducing the error however I was unable to. For reference, this is the code I used: 

X, y = np.random.random((10, 100)), np.random.random((10, 1))
dataset = dc.data.NumpyDataset(X=X, y=y)
model_dir='~/deepchem'
pytorch_model = torch.nn.Sequential(
   torch.nn.Linear(100, 1000),
   torch.nn.Tanh(),
   torch.nn.Linear(1000, 1))
model = dc.models.TorchModel(pytorch_model, loss=dc.models.losses.L2Loss(),model_dir= model_dir, device='cuda')
loss = model.fit(dataset, nb_epoch=5)
model.save_checkpoint()
model_new = dc.models.TorchModel(model_dir= model_dir, model=pytorch_model, loss=dc.models.losses.L2Loss())
model_new.restore()

Hope this is helpful. Thank you.

**Comment by rbharath** [comment]
Going to close this issue. Please feel free to re-open if you're still facing issues
